File: python/secret-handshake/secret_handshake.py
DICT = {'wink': 1, 'double blink': 2, 'close your eyes': 4, 'jump': 8}

def commands(n):
    n = int(n, 2)
    l = [k for k,v in DICT.items() if n & v]
    return l[::-1] if n & 16 else l

File: csharp/secret-handshake/SecretHandshake.cs
using System.Collections.Generic;
using System.Linq;

public static class SecretHandshake
{
    private static readonly IDictionary<int, string> Orders = new Dictionary<int, string>
    {
        { 1, "wink" },
        { 2, "double blink" },
        { 4, "close your eyes" },
        { 8, "jump" }
    };

    public static string[] Commands(int commandValue)
    {
        var s = Orders.Where(o => (commandValue & o.Key) > 0).Select(o => o.Value);

        return (commandValue & 16) > 0 ? s.Reverse().ToArray() : s.ToArray();
    }
}
